src/core/Akka/Util/StringLike.cs

```diff
@@ -26,7 +25,15 @@ public static bool Like(this string text,string pattern, bool caseSensitive = fa
         {
             pattern = pattern.Replace(".", @"\.");
             pattern = pattern.Replace("?", ".");
-            pattern = pattern.Replace("*", ".*?");
+            if(pattern[0] == '*' && pattern[pattern.Length-1]!='*'){
+                pattern = pattern.Replace("*", "(.*?)");
+                pattern = pattern + "(?!.)";
+            }else if(pattern[0]!='*' && pattern[pattern.Length-1]=='*'){
+                pattern = pattern.Replace("*", "(.*?)");
+                pattern = "^"+pattern;
+            }else{
+                pattern = pattern.Replace("*", "(.*?)");
+            }
```

Comment:
Could you please move this into its own PR ? Makes its easier to reason about the changes.

Reply:
Not sure what you mean by PR? Not completely up on the short hand.

Reply:
Soory, PR means Pull Request. In the end it would be good to have a pull request per issue it handles.